### PR TITLE
#746 Modify the child's age in wish cards

### PR DESCRIPTION
**Pull request by furmanp**
Added a condition that checks whether the new date format exists.
If it does, it's used to calculate child's age. If not, the old one is being used.
(assuming either of those is present)

added a line to Contributing.md about a docker config file.

### CODE REVIEW

**Comment by stacysealky** [comment]
@furmanp 

You should add the new birthday logic to handleGetSingle() function in order to correct #746 

`this.renderView(res, 'wishcard/single', {
				wishcard: {
					...wishcard,
					age: wishcard?.childBirthday
						? moment(new Date()).diff(wishcard?.childBirthday, 'years')
						: 'Not Provided',
				},`
Line 427 of wishcard.ts file

File: src/controller/wishcard.ts [hunk 1]
```diff
@@ -116,9 +116,13 @@ export default class WishCardController extends BaseController {
...
 			const data = [] as unknown as WishCard & { age: number }[];
-
+			let birthday: moment.Moment;
 			for (let i = 0; i < wishcards.length; i++) {
-				const birthday = moment(new Date(wishcards[i].childBirthday));
+				if (wishcards[i].childBirthYear) {
+					birthday = moment(new Date(wishcards[i].childBirthYear));
+				} else {
+					birthday = moment(new Date(wishcards[i].childBirthday));
+				}
```
Comment (Enubia) [inline src/controller/wishcard.ts:125]: These changes need to be applied in the handleGetSingle function since that one is called on detail page render

**Review by Enubia — CHANGES_REQUESTED** [review]

File: src/controller/wishcard.ts (hunk 1 above)
Comment (stacysealky) [reply src/controller/wishcard.ts:125]: @Enubia lol I literally wrote that above 😅

File: src/controller/wishcard.ts [hunk 2]
```diff
@@ -116,9 +116,13 @@ export default class WishCardController extends BaseController {
 			const wishcards = await this.wishCardRepository.getAll();
 
 			const data = [] as unknown as WishCard & { age: number }[];
-
+			let birthday: moment.Moment;
 			for (let i = 0; i < wishcards.length; i++) {
-				const birthday = moment(new Date(wishcards[i].childBirthday));
+				if (wishcards[i].childBirthYear) {
```
[the remark this answers is not in the document]
Comment (furmanp) [reply src/controller/wishcard.ts:121]: updated according to the comments

**Review by Enubia — APPROVED** [review]

**Pull request merged**